File: python/all-your-base/all_your_base.py
def rebase(input_base, digits, output_base):
    if input_base <= 1 or output_base <= 1:
        raise ValueError('input base should be larger than one')

    num = 0
    output = []
    for i in range(len(digits)):
        if digits[-1-i] < 0 or digits[-1-i] >= input_base:
            raise ValueError('wrong digit')
        num += digits[-1-i] * input_base ** i
    
    if len(digits) == 0 or num == 0:
        return output

    while True:
        quotient = num // output_base
        remainder = num % output_base

        output.append(remainder)
        num = quotient
        
        if quotient == 0:
            break
    output.reverse()
    return output

File: ruby/all-your-base/all_your_base.rb
module BaseConverter
  def self.convert(input_base, digits, output_base)
    raise ArgumentError if input_base <= 1 || output_base <= 1
    raise ArgumentError if digits.any? { |e| e < 0 || e >= input_base }
    return digits if digits.empty?
    num = digits.reverse
                .map
                .with_index { |e, i| e * input_base ** i }
                .reduce(:+)
    data = []
    until num < output_base
      data << num % output_base
      num = num / output_base
    end
    data << num
    data.reverse
  end
end


p BaseConverter.convert(7, [0, 6, 0], 10)